engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -355,4 +361,66 @@ private static boolean isErrorLevelEnable()
     return false;
   }
 
+  /**
+   * Adds Logger Appender
+   * @param name Appender name
+   * @param className Appender class name
+   * @param args Appender args
+   */
+  public static void addAppender(String name, String className, Object args)
+  {
+    if (getAppendersNames().contains(name)) {
+      logger.warn("A logger appender with the name '{}' exists. Cannot add a new logger appender with the same name", name);
+    } else {
+      try {
+        Constructor<?> constructor = Class.forName(className).getConstructor(Object.class);
```

Comment:
There is only one restriction for Log4j appenders that can be added to APEX dynamically. They must have a constructor with the parameter Object. I guess the typical usage of predefined Log4j appenders can be to extend the appender class to add the constructor with the parameter Object and use the value of the parameter for setting of the appender properties or for a reference to the appender properties.